src/sage/geometry/hyperplane_arrangement/ordered_arrangement.py

```diff
@@ -0,0 +1,648 @@
+
+           This functionality requires the ``sirocco`` package to be installed.
+
+        EXAMPLES::
+
+            sage: # needs sirocco
+            sage: A.<x, y> = OrderedHyperplaneArrangements(QQ)
+            sage: H = A(x, y, x + y)
+            sage: H.projective_fundamental_group()
+            Finitely presented group < x0, x1 |  >
+
+            sage: # needs sirocco sage.graphs
+            sage: A3.<x, y, z> = OrderedHyperplaneArrangements(QQ)
+            sage: H = A3(hyperplane_arrangements.braid(4).essentialization())
+            sage: G3 = H.projective_fundamental_group(); G3.sorted_presentation()
+            Finitely presented group
+            < x0, x1, x2, x3, x4 | x4^-1*x3^-1*x2^-1*x3*x4*x0*x2*x0^-1,
+                                   x4^-1*x2^-1*x4*x2, x4^-1*x1^-1*x0^-1*x1*x4*x0,
+                                   x4^-1*x1^-1*x0^-1*x4*x0*x1,
+                                   x4^-1*x1^-1*x3*x0*x1*x3^-1*x2^-1*x4*x0^-1*x2,
+                                   x3^-1*x2^-1*x1^-1*x0^-1*x3*x0*x1*x2,
+                                   x3^-1*x1^-1*x3*x1 >
+            sage: G3.abelian_invariants()
+            (0, 0, 0, 0, 0)
+            sage: A4.<t1, t2, t3, t4> = OrderedHyperplaneArrangements(QQ)
+            sage: H = A4(hyperplane_arrangements.braid(4))
+            sage: G4 = H.projective_fundamental_group(); G4.sorted_presentation()
+            Finitely presented group
+            < x0, x1, x2, x3, x4 | x4^-1*x3^-1*x2^-1*x3*x4*x0*x2*x0^-1,
+                                   x4^-1*x2^-1*x4*x2, x4^-1*x1^-1*x0^-1*x1*x4*x0,
+                                   x4^-1*x1^-1*x0^-1*x4*x0*x1,
+                                   x4^-1*x1^-1*x3*x0*x1*x3^-1*x2^-1*x4*x0^-1*x2,
+                                   x3^-1*x2^-1*x1^-1*x0^-1*x3*x0*x1*x2,
+                                   x3^-1*x1^-1*x3*x1 >
+            sage: G4.abelian_invariants()
+            (0, 0, 0, 0, 0)
+
+            sage: # needs sirocco
+            sage: L.<t0, t1, t2, t3, t4> = OrderedHyperplaneArrangements(QQ)
+            sage: H = hyperplane_arrangements.coordinate(5)
+            sage: H = L(H)
+            sage: g = H.projective_fundamental_group()
+            sage: g.is_abelian(), g.abelian_invariants()
+            (True, (0, 0, 0, 0))
+            sage: L(t0, t1, t2, t3, t4, t0 - 1).projective_fundamental_group()
+            Traceback (most recent call last):
+            ...
+            TypeError: the arrangement is not projective
+        """
+        K = self.base_ring()
+        if not K.is_subring(QQbar):
+            raise TypeError('the base field is not in QQbar')
+        if not self.is_linear():
+            raise TypeError('the arrangement is not projective')
+        if self._projective_fundamental_group:
+            return self._projective_fundamental_group
+        n = self.dimension()
+        r = len(self)
+        if n == 1:
+            print("Only empty arrangements for dimension ", n - 1)
+            return None
```

Comment:
This should still return something meaningful without a print statement. Since it is the empty arrangement, shouldn't the result just be $\pi_1(\mathbb{CP}^2) = 1$ the trivial group?

Reply:
You are right, changed as for the affine arrangements, actually it is $\pi_1(\mathbb{CP}^0)$, next push.